TPTP/Translate.hs

```diff
@@ -0,0 +1,85 @@
+{- |
+Module      :  ./SoftFOL/Translate.hs
+Description :  utility to create valid identifiers for atp provers
+Copyright   :  (c) Klaus Luettich, Uni Bremen 2005
+License     :  GPLv2 or higher, see LICENSE.txt
+
+Maintainer  :  luecke@informatik.uni-bremen.de
+Stability   :  provisional
+Portability :  portable
+
+collection of functions used by "Comorphisms.SuleCFOL2SoftFOL" and
+ "SoftFOL.ProveSPASS" for the translation of CASL identifiers and axiom labels
+ into valid SoftFOL identifiers -}
+
+module TPTP.Translate
+    ( transId
+    , transSenName
+    , checkIdentifier
+    , CKType (..)
+    ) where
+
+import Data.Char
+import qualified Data.Set as Set
+
+import Common.Id
+import Common.ProofUtils
+
+
+data CKType = CKSort | CKVar | CKPred | CKOp
+
+
+transSenName :: String -> String
+transSenName = transIdString CKSort . concatMap transToSPChar
+
+{- |
+  SPASS Identifiers may contain letters, digits, and underscores only; but
+  for TPTP the allowed starting letters are different for each sort of
+  identifier.
+-}
+checkIdentifier :: CKType -> String -> Bool
+checkIdentifier t str = case str of
+  "" -> False
+  c : _ -> all checkSPChar str && case t of
+    CKVar -> isUpper c -- for TPTP
+    _ -> isLower c
+
+{- |
+Allowed SPASS characters are letters, digits, and underscores.
+Warning:
+Data.Char.isAlphaNum includes all kinds of isolatin1 characters!! -}
+checkSPChar :: Char -> Bool
+checkSPChar c = isAlphaNum c && isAscii c || '_' == c
+
+transId :: CKType -> Id -> Token
+transId t = mkSimpleId . transIdString t . concatMap transToSPChar . show
+
+transIdString :: CKType -> String -> String
+transIdString t str = case str of
+  "" -> error "SoftFOL.Translate.transId: empty string not allowed"
```

Comment:
should the error message not rather point at SoftFOL.Translate.transIdString?

Reply:
Done with d7d74ed83